packages/mask/src/plugins/Pets/hooks/useNfts.ts

```diff
@@ -0,0 +1,97 @@
+import { useEffect, useMemo, useState } from 'react'
+import { useAsync } from 'react-use'
+import { OpenSea } from '@masknet/web3-providers'
+import {
+    useChainId,
+    useCollectibles,
+    useNFTListConstants,
+    ERC721TokenDetailed,
+    isSameAddress,
+    ERC721ContractDetailed,
+    SocketState,
+} from '@masknet/web3-shared-evm'
+import { cloneDeep } from 'lodash-unified'
+import { delay } from '@masknet/shared-base'
+import type { User, FilterContract } from '../types'
+import { Punk3D } from '../constants'
+
+function useInitNFTs() {
+    const nftList = useNFTListConstants()
+    return useMemo(
+        () =>
+            Object.keys(nftList).map((i) => {
+                const value = nftList[i as keyof typeof nftList]
+                return { name: i, contract: value || '', tokens: [] }
+            }),
+        [nftList],
+    )
+}
+
+export function useNFTs(user: User | undefined) {
+    const initContracts = useInitNFTs()
+    const [nfts, setNfts] = useState<FilterContract[]>(initContracts)
+    const chainId = useChainId()
+    const [fetchTotal, setFetchTotal] = useState<ERC721TokenDetailed[]>([])
+    const { data: collectibles, state } = useCollectibles(user?.address ?? '', chainId)
+
+    useEffect(() => {
+        const tempNFTs: FilterContract[] = cloneDeep(initContracts)
+        if (collectibles.length && state === SocketState.done) {
+            const total = [...fetchTotal, ...collectibles]
+            setFetchTotal(total)
+            for (const NFT of total) {
+                const sameNFT = tempNFTs.find((temp) => isSameAddress(temp.contract, NFT.contractDetailed.address))
+                if (!sameNFT) return
+                const glbSupport =
+                    isSameAddress(NFT.contractDetailed.address, Punk3D.contract) && NFT.tokenId === Punk3D.tokenId
+                const item = { ...NFT.info, tokenId: NFT.tokenId, glbSupport }
+                let filterIdx = -1
+                sameNFT.tokens.filter((token, idxToken) => {
+                    const flag = token.tokenId === NFT.tokenId
+                    if (flag) {
+                        filterIdx = idxToken
+                    }
+                    return flag
+                })
```

Comment:
the is side effect logic, may be can use `lodash.findLastIndex` replace it

see https://lodash.com/docs/4.17.15#findLastIndex